src/main/webapp/courseLoader.js

```diff
@@ -194,7 +203,7 @@
 
   async function makeTermPerceptionChart(termDataObject) {
     const average = (list) =>
```

Comment:
Looks like this average function gets used in multiple functions. Probably best to only define it once instead of repeating yourself. Ideally this would be imported from a utilities file or something like that. For here it's fine to just define it once per file instead if you want.